### ISSUE TITLE
Code snippet gets corrupted by tslint

### ISSUE THREAD

**Issue by martinlindhe** [palantir/tslint#3946]
### Bug Report

- __TSLint version__: 5.10.0
- __TypeScript version__: 2.9.1
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
export class MyClass {
    public preload() {
        // uncomment the following line and save the file:
        }
    }

    public update(): void {
        console.log("hi");
    }
}
```

with `tslint.json` configuration:

```json
{
    "defaultSeverity": "error",
    "extends": [
        "tslint:recommended"
    ],
    "rules": {
        "indent": [true, "spaces", 4],
        "linebreak-style": [true, "LF"],
        "max-line-length": [true, 120],
        "object-literal-sort-keys": false,
        "quotemark": [true, "double"],
        "interface-name": [false],
        "arrow-parens": false,
        "no-console": [false]
    }
}
```

#### Actual behavior

Like this animation shows:

![tslint-corrupt-small](https://user-images.githubusercontent.com/181531/41049648-b29f1c0a-69b1-11e8-9c05-5c90aa672212.gif)

#### Expected behavior

Parsing should perhaps fail and not try to fix up such a broken source code.

**Comment by JoshuaKGoldberg** [comment]
This is a subset of #3808. Should TSLint run in the first place if there are syntax errors? If so, should it --fix?
